### PR TITLE
Bump otel collector to 0.40.0

### PR DESCRIPTION
**Pull request by yuriolisa**
Fixes: #584

### CODE REVIEW

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -265,7 +265,7 @@ spec:
               - args:
                 - --metrics-addr=127.0.0.1:8080
                 - --enable-leader-election
-                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:v0.39.0
+                image: ghcr.io/yurioliveirasa/opentelemetry-operator/opentelemetry-operator:v0.40.0
```
Comment (VineethReddy02) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:268]: This `yurioliveirasa` should be `open-telemetry` you should set this in `Makefile` before executing the `bundle`

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -33,11 +33,11 @@ metadata:
     containerImage: quay.io/opentelemetry/opentelemetry-operator
     createdAt: "2020-12-16T13:37:00+00:00"
     description: Provides the OpenTelemetry components, including the Collector
-    operators.operatorframework.io/builder: operator-sdk-v1.13.0+git
+    operators.operatorframework.io/builder: operator-sdk-v1.13.1
```
Comment (VineethReddy02) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:36]: The `operator-sdk` version we are using in CI is `operator-sdk-v1.13.0+git`, Can you switch the version in your local system to thedesired one? Also, the operator-sdk version semantics differ from the source installed. Below is the discussion on the same we had previously:  

https://github.com/open-telemetry/opentelemetry-operator/pull/336#discussion_r665022289

This version change is causing CI to fail.

File: bundle/metadata/annotations.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ annotations:
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: opentelemetry-operator
   operators.operatorframework.io.bundle.channels.v1: alpha
-  operators.operatorframework.io.metrics.builder: operator-sdk-v1.13.0+git
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.13.1
```
Comment (VineethReddy02) [inline bundle/metadata/annotations.yaml:8]: This file should be changed if `operator-sdk` version is fixed.

**Review by VineethReddy02 — CHANGES_REQUESTED** [review]

File: bundle/metadata/annotations.yaml (hunk 3 above)
Comment (yuriolisa) [reply bundle/metadata/annotations.yaml]: Fixed.

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (yuriolisa) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml]: Fixed.

Comment (yuriolisa) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml]: Fixed.

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (yuriolisa) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml]: Fixed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
 Changes by Version
 ==================
 
+0.40.0
+-------------------
+* Bumped OpenTelemetry Collector to v0.39.0
```
Comment (VineethReddy02) [inline CHANGELOG.md:6]: `v0.40.0`

Comment (yuriolisa) [reply CHANGELOG.md]: Done

**Review by VineethReddy02 — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**